ansible_ai_connect/ai/api/serializers.py

```diff
@@ -638,3 +638,46 @@ class TelemetrySettingsRequestSerializer(serializers.Serializer):
         label="OptOut",
         help_text="Indicates whether the Red Hat Organization opts out of telemetry collection.",
     )
+
+
+class ChatRequestSerializer(serializers.Serializer):
+    conversation_id = serializers.UUIDField(
+        format="hex_verbose",
+        required=False,
```

Comment:
I'm surprised `conversation_id` is not mandatory; given this field does not have a _default_.

What other primary key do you have to differentiate on chat from another?

Reply:
Initially I implemented client (React code) to generate a new `conversation_id` for starting a new chat. However, in the OLS query API defines it as an optional parameter and [the lightspeed-service code](https://github.com/openshift/lightspeed-service/blob/7e4a33f075cf488ad1cb9ea4fd23aa3e505b4ee0/ols/app/endpoints/ols.py#L227-L236) is written to generate a new `conversation_id` when it is not given. So the flow will be as follows:

1. When starting a new conversation, the client call the AI Connect services' chat API without `conversation_id`.
2. The chat API calls OLS's query API without  `conversation_id`.
3. The query API returns a response with a newly created `conversation_id`.
4. The chat API forwards the response to the client.
5. The client saves the `conversation_id` given and use it for following conversation

Note: We have not implemented an action to start a new conversation, which PatternFly virtual assistant already has. When that action is implemented, it should clear the saved `conversation_id`.